lib/providers/token.js

```diff
@@ -117,48 +117,56 @@ class TokenProvider {
    * POSTs the instance identity data to Warden and then parses
    * the response to return a token.
    *
-   * @param {function} callback
+   * @returns {Promise}
    */
-  initialize(callback) {
-    this._getDocument()
-      .then((data) => this._sendDocument(data))
-      .then((data) => {
-        this.token = data.data.token;
-
-        callback(null, {
-          lease_id: data.data.token,
-          lease_duration: data.lease_duration,
-          data: {
-            token: this.token
-          }
-        });
-      }).catch((err) => callback(err, null));
+  initialize() {
+    if (this.token) {
+      return Promise.resolve(this.renew());
+    }
+
+    return new Promise((resolve, reject) => {
+      this._getDocument()
+        .then((data) => this._sendDocument(data))
+        .then((data) => {
+          this.token = data.data.token;
+
+          resolve({
```

Comment:
Just return the `{Object}` here instead of `resolve`ing it, and return the result of this `then` chain instead of wrapping it in a promise that does nothing else. Rinse and repeat for all of these...